src/lints/struct_with_no_pub_fields_changed.ron

```diff
@@ -0,0 +1,76 @@
+SemverQuery(
+    id: "struct_with_no_pub_fields_changed",
+    human_readable_name: "public API struct with no public fields is no longer a struct",
+    description: "A struct without pub fields was converted into an enum or union, breaking pattern matching.",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://internals.rust-lang.org/t/rest-patterns-foo-should-match-non-struct-types/21607"),
+    reference: Some(
+        r#"\
+Even if a structs does not expose pub fields, pattern matching like `matches!(value, Example { .. })` is accessible outside their defining crate. \
+Changing such a struct to make it an enum or union will break this pattern match, as their syntax is different.
```

Comment:
Fix `structs -> struct` typo and simplify language a bit ("defining crate" might confuse some users). I also suggest removing "their syntax is different" since it isn't clear what "their" refers to: pattern matches, breaking changes, structs, enums, unions, or some combination of them.
```suggestion
Even if a struct does not expose pub fields, pattern matching like `matches!(value, Example { .. })` is allowed outside the struct's own crate. \
Changing such a struct into an enum or union will break such pattern matching.
```